resharper/resharper-unity/src/CSharp/Daemon/CallGraph/UnityCallGraphUtil.cs

```diff
@@ -16,5 +19,17 @@ public static bool IsFunctionNode(ITreeNode node)
                     return false;
             }
         }
+        
+        public static DaemonProcessKind GetProcessKindForGraph([NotNull] SolutionAnalysisService solutionAnalysisService)
+        {
+            return solutionAnalysisService.Configuration?.Completed?.Value == true
```

Comment:
It is wrong assumption that completed SWA == DaemonProcessKind.GLOBAL_WARNINGS, there are different kinds which possible when SWA completed and you should pass DaemonProcessKind.GLOBAL_WARNINGS to call graph enging only when stage is executed in GlobalWarnings mode.

It will be better to create new enum: "CallGraphPropagateState" with two values "Propagate", "DoNotPropagate" and pass it to IsMarkedByCallGraphRootMarksProvider instead of bool parameter

Reply:
Also, you have method for checking swa state below, do not copy-paste code